tests/bgp/test_bgp_chasssis.py

```diff
@@ -0,0 +1,424 @@
+            cfg_facts = all_cfg_facts[duthost.hostname][0]['ansible_facts']
+            expected_intfs = []
+            if expected_external_intfs:
+                expected_intfs = copy.deepcopy(expected_external_intfs)
+            pytest_assert(len(cfg_facts['VOQ_INBAND_INTERFACE']) == 1,
+                          "Only one Inband interface is supported")
+            if internal:
+                expected_intfs += cfg_facts['VOQ_INBAND_INTERFACE'].keys()
+                expected_intfs *= num_internal_paths
+
+            intfs = []
+            for _, routes in ip_route.items():
+                for route in routes:
+                    for nexthop in route['nexthops']:
+                        if 'recursive' in nexthop and nexthop['recursive']:
+                            continue
+                        intfs += [nexthop['interfaceName']]
+
+            intfs.sort()
+            expected_intfs.sort()
+            pytest_assert(intfs == expected_intfs, "Route is not learned over expected interfaces")
+
+        routes = {}
+        routes[IPV4], routes[IPV6] = build_routes
+
+        def update_routes(exabgp_port, routes, asn=None, withdraw=False):
+            action = 'withdraw' if withdraw else 'announce'
+            for route in routes:
+                update_route(action, ptfhost.mgmt_ip, exabgp_port, route, asn)
+
+        def verify_routes(duthost, routes, neigh_ip_addrs, neigh_intfs,
+                          nexthops_with_inferior_paths, deleted):
+            nexthops = None if deleted else neigh_ip_addrs
+            num_internal_paths = 0
+            if neigh_intfs:
+                num_internal_paths = len(neigh_ip_addrs) - len(neigh_intfs)
+            else:
+                num_internal_paths = len(neigh_ip_addrs)
+
+            internal = bool(num_internal_paths)
+            for route in routes:
+                wait_until(30, 2, verify_bgp_route, duthost, route['prefix'], nexthops,
+                           nexthops_with_inferior_paths)
+                wait_until(30, 2, verify_ip_route, duthost, route['prefix'], nexthops)
+                if not deleted:
+                    verify_nexthop_interface(duthost, route['prefix'],
+                                             expected_external_intfs=neigh_intfs,
+                                             internal=internal,
+                                             num_internal_paths=num_internal_paths)
+
+        def get_neigh_info(dut, exabgp_ports, neigh_ip_addrs, neigh_intfs_by_dut,
+                           ipversion, neigh_index=0):
+            exabgp_port, neigh_ip_address, neigh_intf = \
+                                           get_exabpg_port(dut, ipversion, neigh_index)
+            exabgp_ports.append(exabgp_port)
+            neigh_ip_addrs.append(neigh_ip_address)
+            if not dut in neigh_intfs_by_dut:
+                neigh_intfs_by_dut[dut] = []
+
+            neigh_intfs_by_dut[dut].append(neigh_intf)
```

Comment:
return this value. Updating function args can bring unnecessary side effects, though in this case it is expected. Easier to follow code when values are returned. Ideally function call should not cause any side effects.